middleware/metrics_test.go

```diff
@@ -0,0 +1,37 @@
+package middleware
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRequestMiddlewareInstalled(t *testing.T) {
+	metricsHandler := http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+		metrics := GetRequestMetrics(r.Context())
+		// Slightly worried about this test becoming flakey if the test runs slow.
+		assert.Equal(t, metrics.Start.Format("2006.01.02 15:04:05"), time.Now().Format("2006.01.02 15:04:05"))
```

Comment:
I bet we can time this a bit better by having one test where the internal handler takes no time, and assert it takes less than 5 seconds.

Then we can have another internal handler that takes 6 seconds and assert it takes more than 5 seconds.

Does that sounds easier to deflake?

Reply:
Yeah, I think even less than 2 seconds shouldn't be flakey, given that having it at "equal" with second granularity I couldn't get it to flake any more. Without the format call, it was about 50/50 flakey or not.